pkg/operation/botanist/monitoring.go

```diff
@@ -424,7 +424,9 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
 
 // DeploySeedGrafana deploys the grafana charts to the Seed cluster.
 func (b *Botanist) DeploySeedGrafana(ctx context.Context) error {
-	if b.Shoot.Purpose == gardencorev1beta1.ShootPurposeTesting {
+	// disable monitoring if shoot has purpose testing or monitoring and loki is disabled
+	if b.Shoot.Purpose == gardencorev1beta1.ShootPurposeTesting ||
+		(!gardenlethelper.IsMonitoringEnabled(b.Config) && !gardenlethelper.IsLokiEnabled(b.Config)) {
```

Comment:
Same here: can we combine this into a single reusable check `Operation.WantsGrafana` or something similar?